plugin/trino-opa/src/test/java/io/trino/plugin/opa/OpaAccessControlFilteringUnitTest.java

```diff
@@ -0,0 +1,362 @@
+import io.trino.spi.security.Identity;
+import io.trino.spi.security.SystemSecurityContext;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static com.google.common.collect.ImmutableSet.toImmutableSet;
+import static com.google.common.net.MediaType.JSON_UTF_8;
+import static io.trino.plugin.opa.RequestTestUtilities.assertStringRequestsEqual;
+import static io.trino.plugin.opa.TestHelpers.NO_ACCESS_RESPONSE;
+import static io.trino.plugin.opa.TestHelpers.OK_RESPONSE;
+import static io.trino.plugin.opa.TestHelpers.systemSecurityContextFromIdentity;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+
+public class OpaAccessControlFilteringUnitTest
+{
+    private static final URI OPA_SERVER_URI = URI.create("http://my-uri/");
+    private HttpClientUtils.InstrumentedHttpClient mockClient;
+    private OpaAccessControl authorizer;
+    private final JsonMapper jsonMapper = new JsonMapper();
+    private Identity requestingIdentity;
+    private SystemSecurityContext requestingSecurityContext;
+
+    @BeforeEach
+    public void setupAuthorizer()
+    {
+        this.mockClient = new HttpClientUtils.InstrumentedHttpClient(OPA_SERVER_URI, "POST", JSON_UTF_8.toString(), request -> OK_RESPONSE);
+        this.authorizer = (OpaAccessControl) OpaAccessControlFactory.create(ImmutableMap.of("opa.policy.uri", OPA_SERVER_URI.toString()), Optional.of(mockClient));
+        this.requestingIdentity = Identity.ofUser("source-user");
+        this.requestingSecurityContext = systemSecurityContextFromIdentity(requestingIdentity);
+    }
+
+    @AfterEach
+    public void ensureRequestContextCorrect()
+            throws IOException
+    {
+        for (String request : mockClient.getRequests()) {
+            JsonNode parsedRequest = jsonMapper.readTree(request);
+            assertEquals(parsedRequest.at("/input/context/identity/user").asText(), requestingIdentity.getUser());
+        }
+    }
+
+    @Test
+    public void testFilterViewQueryOwnedBy()
+    {
+        Identity userOne = Identity.ofUser("user-one");
+        Identity userTwo = Identity.ofUser("user-two");
+        List<Identity> requestedIdentities = ImmutableList.of(userOne, userTwo);
+        this.mockClient.setHandler(buildHandler("/input/action/resource/user/user", "user-one"));
```

Comment:
`this.` isn't needed on any of the references to `mockClient` (same in other tests)